config.dev.toml

```diff
@@ -320,6 +320,15 @@ IndexType = "none"
         # Password is the password for a redis SingleFlight lock.
         # Env override: ATHENS_REDIS_PASSWORD
         Password = ""
+
+        [SingleFlight.Redis.LockConfig]
+            # TTL for the lock in seconds. Defaults to 300 seconds (5 minutes).
+            TTL = 300
```

Comment:
I am not understanding this completely, but is there a reason to have TTL and timeout configured separately?
Can you please update the documentation when these should be used?

Reply:
My idea behind having two separate configurations for them:

1. TTL & timeout are two different things, i.e, TTL sets the expiry of the lock whereas timeout sets the context timeout and allows you to cancel retries after timeout.
2. For our use-case we may have higher TTL (15-20mins) but we wouldn't want retries to last that long.

So this might be helpful for those who would like to customize it in a similar way.

Reply:
I'll check & update the relevant docs.

Reply:
Updated the docs.

Reply:
Regarding the following:
Timeout: Time to acquire locks should be in order of miliseconds or a few seconds not minutes. 
I think we should have it to either 1 or 2.

I do not mind if you increase the TTL to 30 minutes as there maybe users with large modules.

Reply:
That's true, yea it makes sense to reduce the timeout as well.

We also have retries and the timeout is not per retry so I think we should at least have enough room to allow the retries - retries are configured to be 1/ sec up-to 10 so that's roughly >=10s, so 10-20s should be good enough. I'll make the default 15s for now.

Reply:
Updated timeout to 15s. Also, updated default TTL to 15 mins.